processing/src/main/java/org/apache/druid/segment/virtual/ExpressionSelectors.java

```diff
@@ -201,6 +201,9 @@ public static DimensionSelector makeDimensionSelector(
 
     if (baseSelector instanceof ConstantExprEvalSelector) {
       // Optimization for dimension selectors on constants.
+      if (plan.is(ExpressionPlan.Trait.NON_SCALAR_OUTPUT)) {
+        return DimensionSelector.multiConstant(Arrays.asList(baseSelector.getObject().asStringArray()), extractionFn);
```

Comment:
Should it pass null instead of a singleton list of null when `baseSelector.getObject().asStringArray()` is null?